lib/cookies.js

```diff
@@ -10,43 +10,30 @@
  * See the License for the specific language governing permissions and limitations under the License.
  *
  */
-
-var util = require('./util');
+var Cookies = require('js-cookie');
 
 function setCookie(name, value, expiresAt) {
-  var expiresText = '';
-  if (expiresAt) {
-    expiresText = ' expires=' + util.isoToUTCString(expiresAt) + ';';
+  var expires = expiresAt || new Date('2038-01-19T03:14:07.000Z');
+
+  if (typeof expires === 'string') {
+    // Possible that the expiresAt value is passed via JSON
+    expires = new Date(expires);
   }
 
-  var cookieText = name + '=' + value + '; path=/;' + expiresText;
-  setCookie._setDocumentCookie(cookieText);
+  Cookies.set(name, value, {
+    expires: expires,
+    path: '/'
+  });
 
-  return cookieText;
+  return getCookie(name);
 }
 
-// Exposed for testing
-setCookie._setDocumentCookie = function(cookieText) {
-  document.cookie = cookieText;
-};
-
-function getCookie(name) {
-  var pattern = new RegExp(name + '=([^;]*)'),
-      matched = getCookie._getDocumentCookie().match(pattern);
-
-  if (matched) {
-    var cookie = matched[1];
-    return cookie;
-  }
+function getCookie() {
+  return Cookies.get.apply(Cookies, arguments);
 }
 
-// Exposed for testing
-getCookie._getDocumentCookie = function() {
-  return document.cookie;
-};
-
-function deleteCookie(name) {
-  setCookie(name, '', '1970-01-01T00:00:00Z');
+function deleteCookie() {
```

Comment:
Great idea, but instead I think I'll modify this function to expect a `name` parameter to keep it consistent with our prior implementation.

One small thing to note - since we're setting/removing cookies for the root path `{ path: '/' }`, we can move this logic inside of the proxy.